### PR TITLE
Feature/login modal

### PR DESCRIPTION
**Pull request by ItaloCovas**
Login UI with Google and Github Sign in.
I have created the branch with "feature" naming, but the commit was with "chore" naming, sorry for the mistake :x

### CODE REVIEW

File: libs/ui/src/lib/components/button/button.tsx [hunk 1]
```diff
@@ -11,6 +11,8 @@ export function Button({ children, button, type, disabled, onClick }: ButtonProp
     primary: 'group relative inline-block font-bold text-black focus:outline-none',
     secondary:
       'font-sans py-2 px-4 rounded cursor-pointer shadow-md text-white font-bold bg-secondary border-secondary-color hover:bg-gradient-to-l hover:from-[#f05a28] hover:to-[#ec008c]',
+    social:
+      'flex justify-center items-center w-full h-10 bg-primary font-bold text-black focus:outline-none transition-colors duration-300 ease-in-out hover:bg-secondary',
```
Comment (Jbnado) [inline libs/ui/src/lib/components/button/button.tsx:15]: I think you can use primary or secondary button instead creating a new button type.
Then instead do a button above another put them in a row side by side.

**Review by PedroLuisBrilhadori — CHANGES_REQUESTED** [review]

File: next-env.d.ts
Comment (PedroLuisBrilhadori) [inline next-env.d.ts:1]: is this file necessary? next-env.d.ts already been generated in `apps/rpjs-site/next-env.d.ts`

Comment (ItaloCovas) [reply next-env.d.ts:1]: It was generated automatically, if you want I can delete it.

File: libs/ui/src/lib/components/button/github.tsx
Comment (PedroLuisBrilhadori) [inline libs/ui/src/lib/components/button/github.tsx:1]: please, export the github button in `buttons/index.ts`

example: 
```ts
export * from './github';
```

File: libs/ui/src/lib/components/button/google.tsx
Comment (PedroLuisBrilhadori) [inline libs/ui/src/lib/components/button/google.tsx:1]: please, export the google button in `buttons/index.ts`

example: 
```ts
export * from './google';
```

File: libs/ui/src/lib/components/login-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
+'use client';
+
+import Image from 'next/image';
+import { Modal } from 'antd';
+import { Dispatch, SetStateAction } from 'react';
+import { GithubButton } from './button/github';
+import { GoogleButton } from './button/google';
```
Comment (PedroLuisBrilhadori) [inline libs/ui/src/lib/components/login-modal.tsx:7]: I think in we can join these two imports. 

example: 
```ts
import { GithubButton, GoogleButton } from './button';
```

File: libs/ui/src/lib/components/login-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+interface LoginModalProps {
+  isOpen: boolean;
+  setIsOpen: Dispatch<SetStateAction<boolean>>;
+}
+
+interface ImageProps {
+  source: string;
+  altText: string;
+}
```
Comment (PedroLuisBrilhadori) [inline libs/ui/src/lib/components/login-modal.tsx:17]: I think this interface should change its name and be in `ui/src/lib/interfaces`

example of `ui/src/lib/interfaces/image.ts`: 
```ts
export interface Image {
  source: string;
  altText: string;
}
```

File: libs/ui/src/lib/components/login-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      return images[Math.floor(Math.random() * 5)];
+    }
+
+    const randomIndex = Math.floor(Math.random() * availableImages.length);
+    const randomImage = availableImages[randomIndex];
+
+    previousImages.push(randomImage);
+
+    return randomImage;
+  }
```
Comment (PedroLuisBrilhadori) [inline libs/ui/src/lib/components/login-modal.tsx:63]: I think this function can be a hook `useRandomImage`. 

### create the file in `ui/src/lib/hooks/random-image.ts`
example file: 
```ts
import { Image } from '../interfaces';

export const useRandomImage = (images: Image[]) => {
  function nextImage() {
    const previousImages: Image[] = [];

    const availableImages = images.filter((image) => !previousImages.includes(image));

    if (availableImages.length === 0) {
      return images[Math.floor(Math.random() * 5)];
    }

    const randomIndex = Math.floor(Math.random() * availableImages.length);
    const randomImage = availableImages[randomIndex];

    previousImages.push(randomImage);

    return randomImage;
  }

  return [nextImage];
};
```
### using hook: 
```ts
  const [nextImage] = useRandomImage(images);
  const randomImage = nextImage();
```

And image array can be substituted  out to of the component. 

Like this: 
```ts
const images = [
  {
    source: '/login.svg',
    altText: 'Garota acessando sua conta em um celular.',
  },
  {
    source: '/login2.svg',
    altText: 'Mulher acessando sua conta em um celular.',
  },
  {
    source: '/login3.svg',
    altText: 'Garoto colocando uma chave dentro de um cadeado, simbolizando a senha.',
  },
  {
    source: '/login4.svg',
    altText: 'Garota acessando sua conta em um celular.',
  },
  {
    source: '/login5.svg',
    altText: 'Garoto e garota acessando suas contas em um celular.',
  },
];

export default function LoginModal({ isOpen, setIsOpen }: LoginModalProps) {
```

File: apps/rpjs-site/public/login.svg
Comment (PedroLuisBrilhadori) [inline apps/rpjs-site/public/login.svg:1]: I think in svg file names should have a better name. We can rename to a short description or a "tittle".

**Review by PedroLuisBrilhadori — CHANGES_REQUESTED** [review]

**Review by PedroLuisBrilhadori — APPROVED** [review]

File: apps/rpjs-site/components/sections/speakers.tsx [hunk 5]
```diff
@@ -31,6 +31,13 @@ const speakers: Array<Speaker> = [
     title: 'Fodao e fodinha',
     media: 'https://github.com/DouglasSerafim',
   },
+  {
+    name: 'Ítalo Covas',
+    role: 'Frontend Developer',
+    image: 'https://github.com/ItaloCovas.png',
+    title: 'Desenvolvedor Frontend na Framework Digital',
+    media: 'https://github.com/ItaloCovas',
+  },
```
Comment (Jbnado) [inline apps/rpjs-site/components/sections/speakers.tsx:40]: Please remove it, here will be the speakers, soon I will recreate our organizers section

File: libs/ui/src/lib/components/button/button.tsx [hunk 6]
```diff
@@ -23,7 +23,8 @@ export function Button({ children, button, type, disabled, onClick }: ButtonProp
           </span>
         </>
       )}
-      {button === 'secondary' && children}
+
+      {button !== 'primary' && children}
```
Comment (Jbnado) [inline libs/ui/src/lib/components/button/button.tsx:27]: Maybe you can revert it, it's easier to understand when is secondary button instead not primary button

**Review by Jbnado — CHANGES_REQUESTED** [review]

File: libs/ui/src/lib/components/button/button.tsx (hunk 6 above)
Comment (ItaloCovas) [reply libs/ui/src/lib/components/button/button.tsx]: Changes were done, if you can review it, thanks for the review man

File: libs/ui/src/lib/components/button/button.tsx [hunk 7]
```diff
@@ -23,7 +23,8 @@ export function Button({ children, button, type, disabled, onClick }: ButtonProp
           </span>
         </>
       )}
-      {button === 'secondary' && children}
+
+      {button !== 'secondary' && children}
```
Comment (PedroLuisBrilhadori) [inline libs/ui/src/lib/components/button/button.tsx:27]: this line is causing an error. 
image
how to fix:
```ts 
 {button === 'secondary' && children}
```

**Review by PedroLuisBrilhadori — CHANGES_REQUESTED** [review]

**Review by PedroLuisBrilhadori — APPROVED** [review]

**Review by Jbnado — APPROVED** [review]
